delta/app/src/main/scala/ch/epfl/bluebrain/nexus/delta/routes/ResourcesRoutes.scala

```diff
@@ -0,0 +1,196 @@
+                }
+              }
+            }
+          },
+          idSegment { idSegment =>
+            concat(
+              operationName(s"$prefixSegment/resources/{org}/{project}/{schema}/{id}") {
+                concat(
+                  (put & pathEndOrSingleSlash) {
+                    authorizeFor(AclAddress.Project(ref), resourcePermissions.write).apply {
+                      (parameter("rev".as[Long].?) & pathEndOrSingleSlash & entity(as[Json])) {
+                        case (None, source)      =>
+                          // Create a resource with schema and id segments
+                          emit(StatusCodes.Created, resources.create(idSegment, ref, schemaSegment, source).map(_.void))
+                        case (Some(rev), source) =>
+                          // Update a resource
+                          emit(resources.update(idSegment, ref, schemaSegmentOpt, rev, source).map(_.void))
+                      }
+                    }
+                  },
+                  (delete & parameter("rev".as[Long]) & pathEndOrSingleSlash) { rev =>
+                    authorizeFor(AclAddress.Project(ref), resourcePermissions.write).apply {
+                      // Deprecate a resource
+                      emit(resources.deprecate(idSegment, ref, schemaSegmentOpt, rev).map(_.void))
+                    }
+                  },
+                  // Fetches a resource
+                  (get & pathEndOrSingleSlash) {
+                    authorizeFor(AclAddress.Project(ref), resourcePermissions.read).apply {
+                      (parameter("rev".as[Long].?) & parameter("tag".as[Label].?)) {
+                        case (Some(_), Some(_)) => emit(simultaneousTagAndRevRejection)
+                        case (Some(rev), _)     => emit(resources.fetchAt(idSegment, ref, schemaSegmentOpt, rev))
+                        case (_, Some(tag))     => emit(resources.fetchBy(idSegment, ref, schemaSegmentOpt, tag))
+                        case _                  => emit(resources.fetch(idSegment, ref, schemaSegmentOpt))
+                      }
+                    }
+                  }
+                )
+              },
+              (pathPrefix("tags") & post & parameter("rev".as[Long]) & pathEndOrSingleSlash) { rev =>
+                authorizeFor(AclAddress.Project(ref), resourcePermissions.write).apply {
+                  entity(as[TagFields]) { case TagFields(tagRev, tag) =>
+                    operationName(s"$prefixSegment/resources/{org}/{project}/{schema}/{id}/tags") {
+                      // Tag a resource
+                      emit(resources.tag(idSegment, ref, schemaSegmentOpt, tag, tagRev, rev).map(_.void))
+                    }
+                  }
+                }
+              }
+            )
+          }
+        )
+      }
+    )
+
+  private def fetchProject(orgUuid: UUID, projectUuid: UUID): Directive1[ProjectResource] =
+    onSuccess(projects.fetch(projectUuid).runToFuture).flatMap {
```

Comment:
Only the ref is interesting here, we should create a method findRefByUuid that return the ref from the uuids.
This way, we only aim for the cache once.
It is something we will need for other resources so we could put it in an object